src/App.tsx

```diff
@@ -1,109 +1,28 @@
 import React from 'react';
+import { PostList } from './components/PostList';
+import { Post } from './types/Post';
 
 import './App.scss';
 
-// import postsFromServer from './api/posts';
-// import commentsFromServer from './api/comments';
-// import usersFromServer from './api/users';
+import postsFromServer from './api/posts';
+import commentsFromServer from './api/comments';
+import usersFromServer from './api/users';
+
+const posts: Post[] = postsFromServer.map(post => {
+  const userFound = usersFromServer.find(user => post.userId === user.id);
+  const commentsFound = commentsFromServer
+    .filter(comment => post.id === comment.postId);
+
+  return {
+    ...post,
+    user: userFound || null,
+    comments: commentsFound,
+  };
+});
```

Comment:
All good. You can split this big function into several (for good practice),  such as getCommentsByPostId and getUserById. 
1. It will be easier to understand.
2. In case of app scaling, you can reuse your functions.
3. In case of objects changing (for example id becomes string, not a number) it will be easier to maintain.